### DISCUSSION TITLE
能不能提供获取Bot设备信息json的方法

### DISCUSSION THREAD

**Discussion by wdvaini** [mamoe/mirai#999] [Ideas]
现在有loadDeviceInfoJson()的方法，让qq登陆时使用指定的设备信息。但是，如果想第一次登陆时自动生成的设备信息以json的形式保存在数据库中（而不是文件），方便以后登陆直接使用。 现在好像没有方法能获取到Bot登陆时使用的设备信息的json文本信息。能否考虑增加这样的方法？

**Comment by Him188** [comment]
现在就支持，仔细看 fileBasedDeviceInfo 实现

**Reply by wdvaini** [reply]
回复的好快，为老大点赞！！
但是，
fileBasedDeviceInfo始终是基于文件存储的形式来保存设备信息的，我想要的是单纯的字符串。以java代码为例：
bot.login();
String  deviceJsonStr = bot.getDeviceInfoJson(); //获取bot使用的设备信息json字符串（ps：mirai现在没有类似功能的方法）

下次登陆时直接使用 loadDeviceInfoJson(deviceJsonStr) ，加载之前的设备信息的json字符串。

**Reply by Him188** [reply]
看 fileBasedDeviceInfo 的内部实现
